src/macros.rs

```diff
@@ -18,19 +18,16 @@ macro_rules! impl_from_ffi_type {
     };
 }
 
+#[macro_export]
 macro_rules! define_custom_string_type {
     ($ffi_type:ident) => {
-        uniffi::custom_type!($ffi_type, String);
-
-        impl UniffiCustomTypeConverter for $ffi_type {
-            type Builtin = String;
-            fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
-                Ok(val.parse::<$ffi_type>()?)
-            }
-
-            fn from_custom(obj: Self) -> Self::Builtin {
-                obj.to_string()
-            }
-        }
+        uniffi::custom_type!($ffi_type, String, {
+            remote,
+            lower: |value: $ffi_type| value.to_string(),
+            try_lift: |value: String| value.parse()
+                .map_err(|_| uniffi::deps::anyhow::Error::msg(
+                    format!("Failed to parse {} from string", stringify!($ffi_type))
+                )),
+        });
```

Comment:
@rustaceanrob if you have time to spot check this item since I think you wrote this original macro (or ALL the macros maybe?!), to see if it makes sense to you as I'm upgrading it per https://mozilla.github.io/uniffi-rs/next/Upgrading.html#custom-types